### PR TITLE
Processing for Mexico dataset: new (daily) & previous (weekly)

### PR DESCRIPTION
**Pull request by tw-chang-git**
PR for including changes for Mexico dataset

### CODE REVIEW

File: tabgpt/data/amazon/data_setup.py [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+from sklearn.preprocessing import OrdinalEncoder
+from IPython import embed
+
+
+class AmazonData(DataFrameLoader):
+    def __init__(self, task_description='amazon'):
+        super().__init__(task_description)
+
+    def setup(self, testset = False):
+        current_dir = os.path.dirname(os.path.abspath(__file__))
```
Comment (umertens) [inline tabgpt/data/amazon/data_setup.py]: remove, can be directly accessed via `self.current_dir`

File: tabgpt/data/amazon/data_setup.py [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+        # Drop due to date: to the actual time range
+        df_final = df_final[df_final["date"] >= date_start].reset_index(drop=True)   # ~1310000
+
+        # Split data
+        df_train = df_final.query('@date_start <= date < @date_split').reset_index(drop=True)
+        df_test = df_final.query('@date_split <= date <= @date_end').reset_index(drop=True)
+
+        # Normalize data
+        df_train["target"] = np.log(1 + df_train["quantity"])
+        df_test["target"] = df_test["quantity"]
```
Comment (umertens) [inline tabgpt/data/amazon/data_setup.py]: remove the two lines, no need anymore to rename the target column to 'target'

File: tabgpt/data/amazon/data_setup.py [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+            "product category",
+            "product name",
+            "holiday_name",
+            "month",
+            "year",
+            "day_of_week"
+        ]
+        numerical_features = [
+            "average daily price",
+            "past target value"
```
Comment (umertens) [inline tabgpt/data/amazon/data_setup.py]: include 'quantity' in list of numerical_features

File: tabgpt/data/amazon/data_setup.py [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+        ]
+
+        # More normalization
```
Comment (umertens) [inline tabgpt/data/amazon/data_setup.py]: remove the 3 lines of normalization. instead do:

`
self.setup_scaler(numerical_features)

self.scale_columns(df_train, mode='train')

self.scale_columns(df_val)
`

File: tabgpt/data/amazon/data_setup.py [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+        # Drop columns
+        df_train = df_train.drop(columns=['product_id'])
+        df_test = df_test.drop(columns=['product_id'])
+
+        self.df_train = df_train
+        self.df_test = df_test
+        self.numerical_features = numerical_features
+        self.categorical_features = categorical_features
+        self.n_features = len(numerical_features + categorical_features)
+        self.target_column = "target"
```
Comment (umertens) [inline tabgpt/data/amazon/data_setup.py]: use  `self.set_target_column(main_target='quantity',additional_ones=False)`

File: tabgpt/data/mexico/data_setup.py [hunk 6]
```diff
@@ -0,0 +1,247 @@
...
+
+        # Drop due to date: to the actual time range
+        df_final = df_final[df_final["date"] >= date_start].reset_index(drop=True)   # to ~2600000
+
+        # Split data
+        df_train = df_final.query('@date_start <= date < @date_split').reset_index(drop=True)
+        df_test = df_final.query('@date_split <= date <= @date_end').reset_index(drop=True)
+
+        # Normalize data
+        df_train["target"] = np.log(1 + df_train["quantity"])
```
Comment (umertens) [inline tabgpt/data/mexico/data_setup.py]: see review in amazon data_setup, so

1. remove target column renaming
2. add actual target column to numerical features
3. use different scaling
4. set target column appropriately

**Review by umertens — CHANGES_REQUESTED** [review]

File: tabgpt/data/amazon/data_setup.py (hunk 4 above)
Comment (tw-chang-git) [reply tabgpt/data/amazon/data_setup.py]: Additionally modify all usages of `df_test` to `df_val` (also on the project/ side) to align with other datasets

File: tabgpt/data/amazon/data_setup.py (hunk 3 above)
Comment (tw-chang-git) [reply tabgpt/data/amazon/data_setup.py]: Also rename the `past target value` to `past quantity`

File: projects/Mexico/mexico_model.py [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+    # features_embeds_test = embedder.embed(n_cols)
+
+    df_train = mexico.df_train
+    df_val = mexico.df_val
+    # df_test = mexico.df_test
+
+
+    train_dataset = TensorDataset(
+        features_embeds_train,
+        torch.tensor(df_train[mexico.target_column].tolist(), dtype=torch.float32)
```
Comment (umertens) [inline projects/Mexico/mexico_model.py]: use `mexico.main_target`

Comment (tw-chang-git) [reply projects/Mexico/mexico_model.py]: Perhaps also changing all other usages of `mexico.target_column` too?

**Review by umertens — APPROVED** [review]

**Comment by umertens** [comment]
Looks good! :)

**Comment by umertens** [comment]
reopened with single commit
